site/src/Librecores/ProjectRepoBundle/RepoCrawler/GitRepoCrawler.php

```diff
@@ -175,11 +179,380 @@ public function updateProject()
             $project->setDateLastActivityOccurred($latestCommit->getDateCommitted());
         }
 
+        // Retrieve the code quality score for the project and persist it in the database
+        $projectMetrics = $this->getCodeQualityScore($project);
+
+        $project->setScore($projectMetrics);
+
         $this->manager->persist($project);
 
         return true;
     }
 
+    /**
```

Comment:
All of this code is copied from `ProjectMetricsProvider.php`. Why don't you inject that class as service into the GitRepoCrawler and use it here?

Reply:
@imphil It took some time to figure out inheritance rules among the classes. I guess this fixes #276